src/event.cc

```diff
@@ -276,17 +269,28 @@ EventScheduler::dump(Packable *out)
     if (last_event_ran)
         out->appendf("Last event to run: %s\n\n", last_event_ran);
 
-    out->appendf("%-25s\t%-15s\t%s\t%s\n",
+    out->appendf("%-25s\t%-15s\t%s\n",
                  "Operation",
                  "Next Execution",
-                 "Weight",
-                 "Callback Valid?");
+                 "Weight");
 
     for (auto *e = tasks; e; e = e->next) {
-        out->appendf("%-25s\t%0.3f sec\t%5d\t %s\n",
-                     e->name, (e->when ? e->when - current_dtime : 0), e->weight,
-                     (e->arg && e->cbdata) ? cbdataReferenceValid(e->arg) ? "yes" : "no" : "N/A");
+        if (e->call->canceled())
+            continue;
+        out->appendf("%-25s\t%0.3f sec\t%5d\n",
+                     e->name, (e->when ? e->when - current_dtime : 0), e->weight);
+    }
+}
+
+bool
+EventScheduler::find(const AsyncCall::Pointer &call) const
+{
+    for (auto *event = tasks; event; event = event->next) {
+        if (event->call == call)
+            return true;
```

Comment:
Currently it is unused, yes. However if I am to implement the other request you have about removing cancelled calls from the event queue this will become a required method.

Reply:
> if I am to implement the other request you have about removing cancelled calls from the event queue this will become a required method.

I am pretty sure that a correct implementation of the cancel() method should not use this method.